waves/wallet/src/wallet.rs

```diff
@@ -308,8 +306,17 @@ fn compute_balances(wallet: &Wallet, txouts: &[TxOut]) -> Vec<BalanceEntry> {
     let grouped_txouts = txouts
         .iter()
         .filter_map(|utxo| match utxo {
-            TxOut::Explicit(explicit) => Some((explicit.asset.0, explicit.value.0)),
-            TxOut::Confidential(confidential) => {
+            TxOut {
+                asset: confidential::Asset::Explicit(asset),
+                value: confidential::Value::Explicit(value),
+                ..
+            } => Some((*asset, *value)),
+            txout => {
+                let confidential = match txout.clone().into_confidential() {
```

Comment:
I see you are calling `.clone().into` a lot. That sounds like you'd want `to_confidential` as well :)

Reply:
I think you're referring to `as_confidential`. I tried to add it to `rust_elements`, but I failed.

Reply:
Not quite. `as_confidential` would imply as signature of `&self -> &T`. That doesn't work because we no longer store an instance of `ConfidentialTxOut` inside `TxOut` but construct it on the fly instead which makes it impossible to return a reference (what would the reference refer to?).

My suggested `to_confidential` would be similar to `into_confidential` but take `&self` instead of `self`. As per Rust's naming convention `to_` implies an allocation which is what we would need to do for `&self -> T`.

Reply:
That makes sense. I guess it's preferable to do all the allocations in a single place, as long as the name of the method indicates what's going on.

Reply:
Also, nothing prevents us from having `into_` and `to_` :)